samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationUtils.java

```diff
@@ -59,7 +58,7 @@ public Latch getLatch(int size, String latchId) {
   public BarrierForVersionUpgrade getBarrier(String barrierId) {
     return new ZkBarrierForVersionUpgrade(barrierId, zkUtils, debounceTimer, zkConfig.getZkBarrierTimeoutMs());
   }
-  @VisibleForTesting
+
```

Comment:
Yeah. I tried to get rid of this method. But the ZkBarrierVersionUpgrade depends on not only the timers but also the zkutils. I didn't want to start refactoring those classes as well. Perhaps you can address it as a part of [SAMZA-1128](https://issues.apache.org/jira/browse/SAMZA-1128)
If it makes sense to address this in that JIRA, let me know. I will add a TODO in the code as well.